diff --git a/package.json b/package.json
old mode 100755
new mode 100644
index e89992c..3a0773e
--- a/package.json
+++ b/package.json
@@ -8,6 +8,7 @@
     "body-parser": "^1.15.2",
     "bower": "^1.7.9",
     "cookie-parser": "^1.4.3",
+    "csurf": "^1.9.0",
     "express": "^4.14.0",
     "express-session": "^1.14.0",
     "gm": "^1.23.0",
diff --git a/snappyServer.js b/snappyServer.js
index 4f1a250..81130a6 100755
--- a/snappyServer.js
+++ b/snappyServer.js
@@ -18,6 +18,8 @@ var bodyParser = require('body-parser');
 var methodOverride = require('method-override');
 var passport = require('passport');
 var session = require('express-session');
+var csrf = require('csurf')
+
 
 var Server = function Server (config) {
     var app, db;
@@ -88,7 +90,6 @@ var Server = function Server (config) {
         process.exit();
     });
 
-
     app.emailSender = app.config.get("emailSender");
     app.jwtSecret = app.config.get("jwtSecret");
     app.confirmUserUrl = app.config.get("confirmUserUrl");
@@ -109,6 +110,10 @@ var Server = function Server (config) {
 
     app.authenticator = require("./src/utils/authenticator")(app);
     app.all(app.config.get('server:routePrefix')+ '*', app.authenticator.authenticate);
+
+    app.csrfValidator = require("./src/utils/csrfValidator")(app);
+    app.use(app.csrfValidator.csrfvalidate);
+
     routes(app,passport);
     app.get('*', function(req, res) {
       res.sendFile(__dirname + '/public/index.html');
diff --git a/src/utils/csrfToken.js b/src/utils/csrfToken.js
new file mode 100644
index 0000000..f8af3d5
--- /dev/null
+++ b/src/utils/csrfToken.js
@@ -0,0 +1,14 @@
+// csrf token
+module.exports  = function(app){
+    console.log('************ ==========')
+    var ATTACHCSRFToken = {
+        attachcsrftoken : function(req, res, next) {
+            console.log('************')
+            res.locals.csrftokenFunction =  req.csrfToken;
+            next()
+        }
+    };
+
+
+    return ATTACHCSRFToken;
+};
diff --git a/src/utils/csrfValidator.js b/src/utils/csrfValidator.js
new file mode 100644
index 0000000..b70be3c
--- /dev/null
+++ b/src/utils/csrfValidator.js
@@ -0,0 +1,16 @@
+// csrf validation
+module.exports  = function(app){
+    var Csrfvalidate = {
+        csrfvalidate : function(err,req, res, next) {
+            if (err.code !== 'EBADCSRFTOKEN') return next(err)
+
+            // handle CSRF token errors here
+            //res.status(403);
+            res.json({"status": "403", "message": "Browse snappyClicks.com", "response": "Try to use snappyClicks.com for you tasks."});
+            //res.send('form tampered with');
+        }
+    };
+
+
+    return Csrfvalidate;
+};
