// promisepay/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = instance;

var _User = require('./components/User');

var _User2 = _interopRequireDefault(_User);

var _Company = require('./components/Company');

var _Company2 = _interopRequireDefault(_Company);

var _BankAccount = require('./components/BankAccount');

var _BankAccount2 = _interopRequireDefault(_BankAccount);

var _CardAccount = require('./components/CardAccount');

var _CardAccount2 = _interopRequireDefault(_CardAccount);

var _PaypalAccount = require('./components/PaypalAccount');

var _PaypalAccount2 = _interopRequireDefault(_PaypalAccount);

var _WalletAccount = require('./components/WalletAccount');

var _WalletAccount2 = _interopRequireDefault(_WalletAccount);

var _Charge = require('./components/Charge');

var _Charge2 = _interopRequireDefault(_Charge);

var _Item = require('./components/Item');

var _Item2 = _interopRequireDefault(_Item);

var _ItemAction = require('./components/ItemAction');

var _ItemAction2 = _interopRequireDefault(_ItemAction);

var _MarketPlace = require('./components/MarketPlace');

var _MarketPlace2 = _interopRequireDefault(_MarketPlace);

var _Transaction = require('./components/Transaction');

var _Transaction2 = _interopRequireDefault(_Transaction);

var _BatchTransaction = require('./components/BatchTransaction');

var _BatchTransaction2 = _interopRequireDefault(_BatchTransaction);

var _DirectDebitAuthority = require('./components/DirectDebitAuthority');

var _DirectDebitAuthority2 = _interopRequireDefault(_DirectDebitAuthority);

var _Fee = require('./components/Fee');

var _Fee2 = _interopRequireDefault(_Fee);

var _Callback = require('./components/Callback');

var _Callback2 = _interopRequireDefault(_Callback);

var _Tool = require('./components/Tool');

var _Tool2 = _interopRequireDefault(_Tool);

var _Token = require('./components/Token');

var _Token2 = _interopRequireDefault(_Token);

var _Requester = require('./Requester');

var _Requester2 = _interopRequireDefault(_Requester);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function instance(config) {
  const requester = new _Requester2.default(config);

  return {
    users: new _User2.default(requester),
    companies: new _Company2.default(requester),
    bankAccounts: new _BankAccount2.default(requester),
    cardAccounts: new _CardAccount2.default(requester),
    paypalAccounts: new _PaypalAccount2.default(requester),
    walletAccounts: new _WalletAccount2.default(requester),
    directDebitAuthority: new _DirectDebitAuthority2.default(requester),
    fees: new _Fee2.default(requester),
    items: new _Item2.default(requester),
    itemActions: new _ItemAction2.default(requester),
    charges: new _Charge2.default(requester),
    marketPlaces: new _MarketPlace2.default(requester),
    transactions: new _Transaction2.default(requester),
    batchTransactions: new _BatchTransaction2.default(requester),
    callbacks: new _Callback2.default(requester),
    tools: new _Tool2.default(requester),
    tokens: new _Token2.default(requester)
  };
}
