// promisepay/dist/components/Item.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _BaseComponent = require('./BaseComponent');

var _BaseComponent2 = _interopRequireDefault(_BaseComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const URI_BASE = '/items';

let Item = class Item extends _BaseComponent2.default {

  // @see https://reference.promisepay.com/#create-item
  create(data) {
    return this.client.post(URI_BASE, data);
  }

  // @see https://reference.promisepay.com/#list-items
  list(_ref) {
    let limit = _ref.limit;
    let offset = _ref.offset;
    let search = _ref.search;
    let disbursement_id = _ref.disbursement_id;

    return this.client.get(URI_BASE, { limit: limit, offset: offset, search: search, disbursement_id: disbursement_id });
  }

  // @see https://reference.promisepay.com/#show-item
  get(id) {
    return this.client.get(`${ URI_BASE }/${ id }`);
  }

  // @see https://reference.promisepay.com/#update-item
  update(id, _ref2) {
    let name = _ref2.name;
    let amount = _ref2.amount;
    let fee_ids = _ref2.fee_ids;
    let description = _ref2.description;
    let buyer_id = _ref2.buyer_id;
    let seller_id = _ref2.seller_id;

    return this.client.patch(`${ URI_BASE }/${ id }`, { name: name, amount: amount, fee_ids: fee_ids, description: description, buyer_id: buyer_id, seller_id: seller_id });
  }

  deleteFees(id) {
    return this.client.delete(`${ URI_BASE }/${ id }/fees`);
  }

  // @see https://reference.promisepay.com/#delete-item
  delete(id) {
    return this.client.delete(`${ URI_BASE }/${ id }`);
  }

  // @see https://reference.promisepay.com/#show-item-status
  status(id) {
    return this.client.get(`${ URI_BASE }/${ id }/status`);
  }

  // @see https://reference.promisepay.com/#show-item-buyer
  getBuyers(id) {
    return this.client.get(`${ URI_BASE }/${ id }/buyers`);
  }

  // @see https://reference.promisepay.com/#show-item-seller
  getSellers(id) {
    return this.client.get(`${ URI_BASE }/${ id }/sellers`);
  }

  // @see https://reference.promisepay.com/#show-item-fees
  getFees(id) {
    return this.client.get(`${ URI_BASE }/${ id }/fees`);
  }

  // @see https://reference.promisepay.com/#show-item-wire-details
  getWireDetails(id) {
    return this.client.get(`${ URI_BASE }/${ id }/wire_details`);
  }

  // @see https://reference.promisepay.com/#show-item-bpay-details
  getBPayDetails(id) {
    return this.client.get(`${ URI_BASE }/${ id }/bpay_details`);
  }

  // @see https://reference.promisepay.com/#list-item-transactions
  getTransactions(id) {
    return this.client.get(`${ URI_BASE }/${ id }/transactions`);
  }

  // @see https://reference.promisepay.com/#list-item-batch-transactions
  getBatchTransactions(id) {
    return this.client.get(`${ URI_BASE }/${ id }/batch_transactions`);
  }
};
exports.default = Item;
