// promisepay/dist/components/Resource.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _lodash = require('lodash');

var _lodash2 = _interopRequireDefault(_lodash);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

let Resource = class Resource {

  constructor(requester, resourceUri, resourceFields, requiredFields) {
    this.client = requester;
    this.resourceUri = resourceUri;
    this.resourceFields = resourceFields;
    this.requiredFields = requiredFields;
  }

  list(_ref) {
    let limit = _ref.limit;
    let offset = _ref.offset;
    let search = _ref.search;

    return this.client.get(this.resourceUri, { limit: limit, offset: offset, search: search });
  }

  get(id) {
    return this.client.get(`${ this.resourceUri }/${ id }`);
  }

  create(fields) {
    return this.client.post(this.resourceUri, _extends({}, _lodash2.default.omit(fields, this.resourceFields)));
  }

  update(id, fields) {
    return this.client.patch(`${ this.resourceUri }/${ id }`, _extends({}, _lodash2.default.omit(fields, this.resourceFields, 'id')));
  }
};
exports.default = Resource;
