// promisepay/dist/components/Transaction.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = undefined;

var _BaseComponent = require('./BaseComponent');

var _BaseComponent2 = _interopRequireDefault(_BaseComponent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const URI_BASE = '/transactions';

let Transaction = class Transaction extends _BaseComponent2.default {

  list(_ref) {
    let limit = _ref.limit;
    let offset = _ref.offset;
    let account_id = _ref.account_id;
    let item_id = _ref.item_id;
    let transaction_type = _ref.transaction_type;
    let transaction_type_method = _ref.transaction_type_method;
    let direction = _ref.direction;

    return this.client.get(URI_BASE, {
      limit: limit, offset: offset, account_id: account_id, item_id: item_id, transaction_type: transaction_type, transaction_type_method: transaction_type_method, direction: direction
    });
  }

  get(id) {
    return this.client.get(`${ URI_BASE }/${ id }`);
  }

  getUser(id) {
    return this.client.get(`${ URI_BASE }/${ id }/users`);
  }

  getFees(id) {
    return this.client.get(`${ URI_BASE }/${ id }/fees`);
  }

  getWalletAccount(id) {
    return this.client.get(`${ URI_BASE }/${ id }/wallet_accounts`);
  }

  getBankAccount(id) {
    return this.client.get(`${ URI_BASE }/${ id }/bank_accounts`);
  }

  getCardAccount(id) {
    return this.client.get(`${ URI_BASE }/${ id }/card_accounts`);
  }

  getPaypalAccount(id) {
    return this.client.get(`${ URI_BASE }/${ id }/paypal_accounts`);
  }
};
exports.default = Transaction;
